cmd/keeper/keeper.go

```diff
@@ -1754,3 +1762,51 @@ func keeper(c *cobra.Command, args []string) {
 
 	lockFile.Close()
 }
+
+func (p *PostgresKeeper) generateHBA(cd *cluster.ClusterData, role common.Role, userHba []string) []string {
+
+	// Dynamically generate hba auth from clusterData
+	var hbaBase []string
+
+	if p.defaultReplAccess == "all" {
+		// allow for all servers su access
+		hbaBase = append(
+			hbaBase,
+			fmt.Sprintf("host all %s %s %s\n", p.pgSUUsername, "0.0.0.0/0", p.pgSUAuthMethod),
+			fmt.Sprintf("host all %s %s %s\n", p.pgSUUsername, "::0/0", p.pgSUAuthMethod),
+		)
+		// allow only for master server replication access
+		if role == common.RoleMaster {
```

Comment:
This is a change from the current behaviour and it's breaking the standby cluster feature and related tests. So it should be removed.

Reply:
I don't understand why it breaks. 
Master pg_hba should only allow replication user to connect from standby servers.
Disbaled or enabling replication user on standby server should not have any effects. 
When a standby server is promoted, its pg_hba should be automatically updated and would contain the right rules.

Reply:
Stolon can also be in standby cluster mode. In this case the master keeper manages a standby instance that will sync with an external instance. So the other keepers will sync from it (cascading replication) and the above check will break.

I suggested to just remove it because to keep backward compatibility with "all" mode we have to keep the previous rules.

Reply:
Ok, changes done. It should be ok !